bound/bound.go

```diff
@@ -4,7 +4,37 @@
 
 package bound
 
+import "math"
+
 // Bound represents [Min, Max] bounds.
 type Bound struct {
 	Min, Max float64
 }
+
+// Intersection returns a intersected bound between bounds.
+// if len(bounds) == 0 then it will return NaN value bound.
+// Or if there is no intersection bound then it will return Nan value bound.
+func Intersection(bounds ...Bound) Bound {
+	if len(bounds) == 0 {
+		return Bound{Min: math.NaN(), Max: math.NaN()}
+	}
+
+	if len(bounds) == 1 {
+		return bounds[0]
+	}
+
+	ret := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
+	for _, b := range bounds {
```

Comment:
This would be cleaner just sweeping over the bounds to do the min and max, with one final check at the end for the bounds matching. It's probably better too, as it'll be faster in the common case of the set actually having a non-empty intersection.